File: sdk/c#/PicarioXPO.RenderAPI/Fluent/FluentXpoUrlObject.cs
using PicarioXPO.RenderAPI.FluentInterface;

namespace PicarioXPO.RenderAPI.Fluent
{
    /// <summary>
    /// Used by FluentXPOUrlGenerator for object configuration
    /// </summary>
    public sealed class FluentXpoUrlObject : IFluentInterface
    {
        /// <summary>
        /// Returns the wrapped XpoObject
        /// </summary>
        public XpoUrlObject XpoObject
        {
            get
            {
                return xpoObject;
            }
        }

        private XpoUrlObject xpoObject;

        /// <summary>
        /// Creates a new instance of the FluentXpoObject
        /// </summary>
        public FluentXpoUrlObject()
        {
            xpoObject = new XpoUrlObject();
        }

        /// <summary>
        /// Sets the index for this object
        /// </summary>
        public FluentXpoUrlObject SetIndex(int index)
        {
            xpoObject.Index = index;

            return this;
        }

        /// <summary>
        /// Specifies object as design
        /// </summary>
        public FluentXpoUrlDesign Design(string fileName)
        {
            xpoObject.ObjectType = XpoUrlObjectTypes.Design;

            var fluentXpoUrlDesign = new FluentXpoUrlDesign(fileName);

            xpoObject.Design = fluentXpoUrlDesign.XpoUrlDesign;

            return fluentXpoUrlDesign;
        }

        /// <summary>
        /// Specifies object as text
        /// </summary>
        public FluentXpoUrlText Text(string text)
        {
            var fluentXpoUrlText = new FluentXpoUrlText(text);

            xpoObject.Text = fluentXpoUrlText.XpoUrlText;

            return fluentXpoUrlText;
        }

        /// <summary>
        /// Specifies object as color
        /// </summary>
        public FluentXpoUrlColor Color(string color)
        {
            var fluentXpoUrlColor = new FluentXpoUrlColor(color);

            xpoObject.Color = fluentXpoUrlColor.XpoUrlColor;

            return fluentXpoUrlColor;
        }
    }
}

File: sdk/js/PicarioXPO.RenderAPI.V2/UrlGenerators/FluentXpoUrlObject.ts
class FluentXpoUrlObject {
    private xpoObject: XpoUrlObject;

    /*
     * Creates a new instance of the FluentXpoObject
     */
    constructor() {
        this.xpoObject = new XpoUrlObject();
    }
    
    /*
     * Returns the wrapped XpoObject
     */
    getXpoObject(): XpoUrlObject { return this.xpoObject; }

    /*
     * Sets the index for this object
     */
    setIndex(index: number) {
        this.xpoObject.setIndex(index);

        return this;
    }

    /*
     * Specifies object as design
     */
    design(fileName: string) {
        this.xpoObject.setObjectType(UrlGeneratorModule.XpoUrlObjectTypes.Design);
        var fluentXpoUrlDesign = new FluentXpoUrlDesign(fileName);
        this.xpoObject.setDesign(fluentXpoUrlDesign.xpoUrlDesign);

        return fluentXpoUrlDesign;
    }

    /*
     * Specifies object as text
     */
    text(text: string) {
        var fluentXpoUrlText = new FluentXpoUrlText(text);
        this.xpoObject.setText(fluentXpoUrlText.getXpoUrlText());

        return fluentXpoUrlText;
    }

    /*
     * Specifies object as color
     */
    color(color: string) {
        var fluentXpoUrlColor = new FluentXpoUrlColor(color);
        this.xpoObject.setColor(fluentXpoUrlColor.getXpoUrlColor());

        return fluentXpoUrlColor;
    }
 }